prototype2/common/Socket.cpp

```diff
@@ -17,6 +17,25 @@
 #define SEND_FLAGS 0
 #endif
 
+
+bool Socket::isValidIp(std::string ipAddress) {
+  struct sockaddr_in sa;
+  int result = inet_pton(AF_INET, ipAddress.c_str(), &(sa.sin_addr));
+  return result != 0;
+}
+
+std::string Socket::getHostByName(std::string &name) {
+  struct hostent * hp = gethostbyname(name.c_str());
+  if (hp == NULL) {
```

Comment:
```suggestion
  if (hp == nullptr) {
```
Use of `NULL` is not recommended as it is defined through `#define NULL 0`. Use `nullptr` as it makes the compiler do type checking.